CMake generating failed in macOS with OPEN_FOR_IDE flag set

```bash
cmake -G Ninja -DOPEN_FOR_IDE=1 ../foundationdb
```
The command above failed with `fdb_c_options` target missing:

```bash
CMake Error at fdbclient/CMakeLists.txt:189 (add_dependencies):
  The dependency target "fdb_c_options" of target "fdbclient_sampling" does
  not exist.


CMake Error at fdbclient/CMakeLists.txt:182 (add_dependencies):
  The dependency target "fdb_c_options" of target "fdbclient" does not exist.
```

--- a/bindings/CMakeLists.txt
+++ b/bindings/CMakeLists.txt
@@ -1,6 +1,6 @@
+add_subdirectory(c)
 if(NOT OPEN_FOR_IDE)
   # flow bindings currently doesn't support that
-  add_subdirectory(c)
   add_subdirectory(flow)
 endif()
 if(WITH_PYTHON_BINDING)

--- a/bindings/c/CMakeLists.txt
+++ b/bindings/c/CMakeLists.txt
@@ -130,9 +130,14 @@ if(NOT WIN32)
     target_link_libraries(fdb_c90_test PRIVATE fdb_c)
   endif()
 
+  if(OPEN_FOR_IDE)
+    set(FDB_C_TARGET $<TARGET_OBJECTS:fdb_c>)
+  else()
+    set(FDB_C_TARGET $<TARGET_FILE:fdb_c>)
+  endif()
   add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/libfdb_c.so
-    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:fdb_c> ${CMAKE_CURRENT_BINARY_DIR}/libfdb_c.so
+    COMMAND ${CMAKE_COMMAND} -E copy ${FDB_C_TARGET} ${CMAKE_CURRENT_BINARY_DIR}/libfdb_c.so
     DEPENDS fdb_c
     COMMENT "Copy libfdb_c to use as external client for test")
   add_custom_target(external_client DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/libfdb_c.so)

